### PR TITLE
EN6-206 Implementing clone resource endpoint

### PR DESCRIPTION
**Pull request by BrenoQVDS**
Adding new endpoint that clone resources
/plugins/cms/assets/{resourceId}/clone

### CODE REVIEW

**Comment by entando-service** [comment]
All is Well

File: plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/resource/IResourceManager.java [hunk 1]
```diff
@@ -128,6 +128,15 @@
...
     public void updateResource(ResourceDataBean bean) throws ApsSystemException;
 
+    /**
+     * Clone una risorsa nel db, indipendentemente dal tipo.
+     *
+     * @param resource La risorsa da clonare.
+     * @return la risorsa aggiunta.
+     * @throws ApsSystemException in caso di errore.
+     */
+    public ResourceInterface cloneResource(ResourceInterface resource) throws ApsSystemException;
```
Comment (ffleandro) [inline plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/resource/IResourceManager.java]: You should not change ResourceManager for this feature.
All cloning business logic should be inside ResourcesService and then just call ResourceManager.addResource

File: plugins/entando-plugin-jacms/src/main/java/org/entando/entando/plugins/jacms/aps/system/services/resource/ResourcesService.java [hunk 2]
```diff
@@ -149,13 +165,25 @@ public void deleteAsset(String resourceId) {
         }
     }
 
-    public AssetDto editAsset(String resourceId, MultipartFile file, String description, List<String> categories) {
+    private ResourceInterface loadResource(String resourceId) {
```
Comment (ffleandro) [inline plugins/entando-plugin-jacms/src/main/java/org/entando/entando/plugins/jacms/aps/system/services/resource/ResourcesService.java]: Method should be called getAsset.
Also, resourceInterface should never be exposed, even internally. Use AssetDto instead.

**Comment by entando-service** [comment]
All is Well

**Comment by entando-service** [comment]
All is Well

**Pull request merged**